File: python-vsc/problems/66.加一.py
#
# @lc app=leetcode.cn id=66 lang=python3
#
# [66] 加一
#

# @lc code=start
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        if not digits:
            return digits
        digits_len = len(digits)
        for j in range(digits_len - 1, -1, -1):
            if digits[j] < 9:
                digits[j] += 1
                return digits
            else:
                digits[j] = 0 
        return [1] + digits


# @lc code=end


# class Solution:
#     def plusOne(self, digits: List[int]) -> List[int]:
#         if not digits:
#             return digits
#         digits_len = len(digits)
#         for j in range(digits_len - 1, -1, -1):
#             if digits[j] < 9:
#                 digits[j] += 1
#                 return digits
#             else:
#                 digits[j] = 0 
#         return [1] + digits

File: java-vsc/problems/66.加一.java
/*
 * @lc app=leetcode.cn id=66 lang=java
 *
 * [66] 加一
 */

// @lc code=start
class Solution {
    public int[] plusOne(int[] digits) {
        int length = digits.length;
        for (int i = length - 1; i >= 0; i--) {
            if (digits[i] < 9) {
                digits[i]++;
                return digits;
            }
            digits[i] = 0;
        }
        int []result = new int[length+1];
        result[0] = 1;
        for (int i = 0; i < length; i++) {
            result[i+1] = digits[i];
        }
        return result;
    }
}
// @lc code=end

